test/integration/websocket_integration_test.cc

```diff
@@ -133,6 +133,15 @@ void WebsocketIntegrationTest::initialize() {
         [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
                 hcm) -> void { hcm.mutable_http2_protocol_options()->set_allow_connect(true); });
   }
+  config_helper_.addFilter(R"EOF(
+name: envoy.filters.http.compressor
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.compressor.v3.Compressor
+  compressor_library:
+    name: test
+    typed_config:
+      "@type": type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip
+)EOF");
```

Comment:
Would it make sense to add a similar change to the http2 integration test? To make sure compression actually works for h2?

Reply:
While my intuition is it shouldn't be done, I don't see anything in either spec about legality of compressing HTTP/1 connect payload which makes me wonder if it's legal, and if it is legal if we should support it by default.

https://tools.ietf.org/html/rfc7540#section-8.3
https://tools.ietf.org/html/rfc8441

@PiotrSikora @lizan thoughts?

Reply:
cc @DavidSchinazi

Reply:
actually @rgs1 since you reported the issue in the first place, do you know if it broke WS on HTTP/1, HTTP/2 or both?  If it's not supported globally it'd be a useful data point.

Reply:
on HTTP/1 (downstream & upstream)

Reply:
Fair.  Either way I agree that integration tests for HTTP/2 would be good to have.

Reply:
Transfer-Encoding is not allowed on CONNECT responses. [RFC 7230 s3.3.1](https://tools.ietf.org/html/rfc7230#section-3.3.1) states:
> A server MUST NOT send a Transfer-Encoding header field in any 2xx (Successful) response to a CONNECT request

Reply:
Sorry. i little bit lost here: are we talking about H/2 integration test for WS or ? If former - i think initializer (where this new compression config is added) is protocol agnostic and most likely used in all of the integration tests (and i assume we do have h/2 ws)

Reply:
Yeah, if it runs for HTTP/1 and HTTP/2 you're fine here.
Unfortunately I think including extensions in core tests is legal so I think you're going to have to add similar test to the compressor lib.

Reply:
actually looks like we do not have h2 integration tests for websockets. Looks like you were about to add them in cd171d9a95f551628453c83f0b00924e8b310bf0 (at least this is where WebsocketTestParams struct were introduced) but it is actually not being used anywhere so all h2 ws upgrade path is untested right now